web/src/app/util/QuerySetFavoriteButton.js

```diff
@@ -52,44 +50,24 @@ export function QuerySetFavoriteButton(props) {
     typeName = 'schedule'
     id = props.scheduleID
   } else {
-    return null
+    throw new Error('unknown type')
   }
-  return (
-    <Query
-      query={queries[typeName]}
-      variables={{ id: id }}
-      render={({ data }) => {
-        if (!data || !data[typeName]) return null
-
-        return renderMutation(data[typeName].isFavorite, id, typeName)
-      }}
-    />
-  )
-}
-
-function renderMutation(isFavorite, id, typeName) {
-  return (
-    <Mutation mutation={mutation} refetchQueries={[`${typeName}FavQuery`]}>
-      {mutation => renderSetFavButton(isFavorite, mutation, id, typeName)}
-    </Mutation>
-  )
-}
+  const { data, loading } = useQuery(queries[typeName], {
```

Comment:
We could probably not show it if there's an error, and back off/start retrying every 30s (or whatever we have that global const set to) and have it appear once/if it is ever successful. What do you think? I don't think it warrants a dialog or a popup

Reply:
Sorry, I meant only retrying the querying of the component to show on the page, not the mutation where the user sets it to true or false

Reply:
What do you think of changing it to an error icon with title/alt-text as the error message?

It might be confusing if the query were to break and fav icons just stopped showing up. We could probably do the same if you click to set it and that fails.

Reply:
^ let's do this in a separate PR on the feature branch (before merging to master)